client/block.c

```diff
@@ -1992,15 +1985,95 @@ void xdag_block_finish()
 	pthread_mutex_lock(&block_mutex);
 }
 
-int xdag_get_block_info(xdag_hash_t hash, void *data, int (*callback)(void*, int, xdag_hash_t, xdag_amount_t, xdag_time_t, xdag_remark_t))
+int xdag_get_block_info(xdag_hash_t hash, void *data, int (*callback)(void*, int, xdag_hash_t, xdag_amount_t, xdag_time_t, const xdag_remark_t))
 {
 	pthread_mutex_lock(&block_mutex);
 	struct block_internal *bi = block_by_hash(hash);
 	pthread_mutex_unlock(&block_mutex);
 
 	if(callback) {
-		return callback(data, bi->flags & ~BI_OURS,  bi->hash, bi->amount, bi->time, bi->remark ? bi->remark : "");
+		return callback(data, bi->flags & ~BI_OURS,  bi->hash, bi->amount, bi->time, ((bi->flags & BI_REMARK) && bi->bremark != NULL) ? (const char*)bi->bremark->remark : "" );
 	}
 
 	return 0;
 }
+
+size_t remark_acceptance(struct block_internal* bi, xdag_remark_t origin, xdag_remark_t strbuf)
```

Comment:
Bugfix buffer overflow
Rechecking if the string is good, check it only when it is sent is useless, someone can send a bad string bypassing the check while sending it.